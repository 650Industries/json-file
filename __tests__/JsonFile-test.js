'use strict';

let JsonFile = require('../JsonFile');

describe('JsonFile', () => {
  it(`is a class`, () => {
    let file = new JsonFile('../package.json');
    expect(file instanceof JsonFile).toBe(true);
  });

  it(`has static functions`, () => {
    expect(JsonFile.readAsync).toBeDefined();
    expect(JsonFile.writeAsync).toBeDefined();
  });

  pit(`reads JSON from a file`, () => {
    // require('fs');
    let file = new JsonFile('./package.json');
    return file.readAsync().then(object => {
      expect(object.version).toBeDefined();
    });
  });
});
