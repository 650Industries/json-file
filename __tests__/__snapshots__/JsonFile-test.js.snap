// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`has useful error messages for JSON parsing errors 1`] = `
"Error parsing JSON file: /Users/ville/Projects/expo/universe/libraries/json-file/__tests__/files/syntax-error.json
└─ Cause: SyntaxError: Unexpected string in JSON at position 602
  23 |     \\"ios\\": {
  24 |       \\"supportsTablet\\": true,
> 25 |       \\"bundleIdentifier\\" \\"com.wow.testapp\\"
     |                          ^
  26 |     }
  27 |   }
  28 | }"
`;

exports[`has useful error messages for JSON5 parsing errors 1`] = `
"Error parsing JSON file: /Users/ville/Projects/expo/universe/libraries/json-file/__tests__/files/syntax-error.json5
└─ Cause: SyntaxError: JSON5: invalid character ',' at 4:15
  2 |   expo: {
  3 |     // The following line has an error
> 4 |     sdkVersion, \\"26.0.0\\",
    |               ^
  5 |   },
  6 | }
  7 | "
`;
